hazelcast/src/test/java/com/hazelcast/cp/internal/CPGroupRebalanceTest.java

```diff
@@ -86,6 +90,43 @@ public void test() throws Exception {
                 assertEquals(leadershipsString(leadershipsMap), leadershipsPerMember, count);
             }
         });
+
+        // terminate a member and check if the leadership rebalance process did occur
+        HazelcastInstance terminatedMember = instances[new Random().nextInt(instances.length)];
+        UUID termindatedMemberUuid = terminatedMember.getCPSubsystem().getLocalCPMember().getUuid();
+        terminatedMember.getLifecycleService().terminate();
+
+        HazelcastInstance[] aliveInstances = Arrays.stream(instances)
+                .filter(instance -> instance != terminatedMember)
+                .toArray(HazelcastInstance[]::new);
+
+        waitAllForSafeState(aliveInstances);
+
+        HazelcastInstance newMetadataLeader = getLeaderInstance(aliveInstances, getMetadataGroupId(aliveInstances[0]));
+        Collection<CPMember> aliveCpMembers = newMetadataLeader.getCPSubsystem()
+                .getCPSubsystemManagementService()
+                .getCPMembers()
+                .toCompletableFuture()
+                .get().stream()
+                .filter(cpMember -> !cpMember.getUuid().equals(termindatedMemberUuid))
+                .collect(Collectors.toList());
+
+        // Assert eventually since during the test
+        // a long pause can cause leadership change unexpectedly.
+        assertTrueEventually(() -> {
+            // Wait for leader election all groups
+            for (CPGroupId groupId : groupIds) {
+                waitAllForLeaderElection(aliveInstances, groupId);
+            }
+
+            rebalanceLeaderships(newMetadataLeader);
+
+            Map<CPMember, Collection<CPGroupId>> leadershipsMap = getLeadershipsMap(newMetadataLeader, aliveCpMembers);
+            for (Entry<CPMember, Collection<CPGroupId>> entry : leadershipsMap.entrySet()) {
+                int count = entry.getValue().size();
+                assertEquals(leadershipsString(leadershipsMap), 25, count);
```

Comment:
```suggestion
                assertEquals(leadershipsString(leadershipsMap), (groupCount - 1) / groupSize , count);
```
One minor comment to help the reader to understand where `25` come from.